addon/services/activity-tracking.ts

```diff
@@ -0,0 +1,82 @@
+import { debounce } from '@ember/runloop';
+import Service, { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+import Configuration from '@upfluence/hyperevents/configuration';
+
+export type ActivityType = 'page_view' | 'button_click' | 'component_view';
+export type Activity = {
+  type: ActivityType;
+  action?: string;
+  path: string;
+  route: string;
+  origin?: string;
+  version?: string;
+  extra?: {};
+};
+
+const RETRY_ATTEMPTS: number = 1;
+const THROTTLE_TIME_MS = 1000;
+
+export default class ActivityTracking extends Service {
```

Comment:
Considering the following fields are common to all activities, can we merge them into the Activity in the service itself instead of passing them from outside every time? 

- origin
- route
- path
- version

This way we only have to handle `type` & `action` in the different decorators/helpers and reduce the code duplication :)

Reply:
Agreed, and that probably solves what Alexis was saying here -> https://github.com/upfluence/hyperevents/pull/7#discussion_r1160877225